### PR TITLE
Feat: Upgrade clickhouse server docker image

### LINKED ISSUE #30: In Redwood, we should use the latest version of Clickhouse
Hi,

I'm trying to add a new connector to a third party provider with cairn clickhouse. However, to make this work we need to upgrade the clickhouse server version used.

`FROM docker.io/yandex/clickhouse-server:22.1.3.7`

However, I checked this repo in docker hub and it seems like there are no latest image of clickhouse released. Please upgrade the version to latest.

### PR DESCRIPTION
**Pull request by FahadKhalid210**
fix #30 
Upgrade clickhouse docker image and fix query issues due to deprecation of[ Live views](https://clickhouse.com/docs/en/sql-reference/statements/create/view#live-view-deprecated).

We currently have 2 live views **course_enrollments** & **course_block_completion**. After upgrading clickhouse version, we encountered errors due to deprecation of live views.

image

After transitioning from live views to normal views to resolve the issue, we encountered a new challenge: `course_enrollment` & `course_block_completion` tables are only accessible in main db(openedx) and we get error on staff specified db’s. It's because staff user's don't have access to the tables that view statement is utilizing(`_openedx_course_enrollments`, `_openedx_user_profiles` & `_openedx_users` etc). 

To resolve this, we've extended access to these tables for staff users and updated the corresponding row policy. This decision is backed by the fact that the live view already displays data from these tables through join operations.

### CODE REVIEW

File: tutorcairn/templates/cairn/apps/clickhouse/migrations.d/0012_modify_course_enrollments.sql [hunk 1]
```diff
@@ -0,0 +1,23 @@
+RENAME TABLE _openedx_course_enrollments TO openedx_course_enrollments;
+RENAME TABLE _openedx_user_profiles TO openedx_user_profiles;
+RENAME TABLE _openedx_users TO openedx_users;
+
+DROP TABLE course_enrollments;
```
Comment (DawoudSheraz) [inline tutorcairn/templates/cairn/apps/clickhouse/migrations.d/0012_modify_course_enrollments.sql:5]: why do we need to drop the table here? Shouldn't it be view (considering view is being created right after that)?

File: tutorcairn/templates/cairn/build/cairn-clickhouse/Dockerfile [hunk 2]
```diff
@@ -1,11 +1,11 @@
...
-FROM docker.io/yandex/clickhouse-server:22.1.3.7
+# https://hub.docker.com/r/clickhouse/clickhouse-server/tags
+FROM docker.io/clickhouse/clickhouse-server:24.1.8.22
 
 # The clickhouse repo is currently unavailable in some parts of the world. If we don't
 # remove this repo here then `apt update` will fail. Check if the problem is resolved with:
 # curl https://repo.yandex.ru/clickhouse/deb/stable/
 # The above command should be a 200, and not a 404.
-RUN rm /etc/apt/sources.list.d/clickhouse.list
+# RUN rm /etc/apt/sources.list.d/clickhouse.list
```
Comment (DawoudSheraz) [inline tutorcairn/templates/cairn/build/cairn-clickhouse/Dockerfile]: nit: remove this if it is not needed.

File: tutorcairn/templates/cairn/apps/clickhouse/migrations.d/0012_modify_course_enrollments.sql (hunk 1 above)
Comment (FahadKhalid210) [reply tutorcairn/templates/cairn/apps/clickhouse/migrations.d/0012_modify_course_enrollments.sql:5]: Table is already created using **Live View** command, that's why we need to drop it.

File: tutorcairn/templates/cairn/apps/clickhouse/migrations.d/0012_modify_course_enrollments.sql [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    openedx_users.email AS user_email,
+    openedx_user_profiles.year_of_birth AS user_year_of_birth,
+    openedx_user_profiles.gender AS user_gender,
+    openedx_user_profiles.level_of_education AS user_level_of_education,
+    openedx_user_profiles.city AS user_city,
+    openedx_user_profiles.state AS user_state,
+    openedx_user_profiles.country AS user_country
+FROM openedx_course_enrollments
+INNER JOIN openedx_user_profiles ON openedx_course_enrollments.user_id = openedx_user_profiles.user_id
+INNER JOIN openedx_users ON openedx_course_enrollments.user_id = openedx_users.id;
```
Comment (DawoudSheraz) [inline tutorcairn/templates/cairn/apps/clickhouse/migrations.d/0012_modify_course_enrollments.sql]: nit: end of file line missing

**Review by DawoudSheraz — APPROVED** [review]

**Pull request merged**